src/octoprint/util/comm.py

```diff
@@ -2608,11 +2669,13 @@ def _gcode_M25_queuing(self, cmd, cmd_type=None, gcode=None, subcode=None, *args
 			self.setPause(True)
 
 	def _gcode_M28_sent(self, cmd, cmd_type=None, gcode=None, subcode=None, *args, **kwargs):
+		self._waitingForM28Confirmation = True
```

Comment:
I fear a race condition here. Since sending and receiving are handled in separate threads, if the stars align just right the `_sent` handler might only be triggered after the monitor thread has already received and parsed the accompanying `ok`. Which means the `ok` for the `M28` won't trigger setting the flag, only the next ok will. Not sure if that is a problem in actual operation.

And no, there is sadly no way at the moment to match up sent commands with received oks (long story short: bug in several firmware variants out there makes commands not get acknowledged in the order they were sent in).

Reply:
yes, it's not a good solution to use switches like this... I try to explain more in the comment where the variable is defined.